components/doc.tsx

```diff
@@ -421,23 +432,71 @@ declare global {
   }
 }
 
-export function Usage({ children }: { children: Child<string> }) {
+export function Usage(
+  { children, item, isType }: {
+    children: Child<string>;
+    item?: string;
+    isType?: boolean;
+  },
+) {
   const url = take(children);
   const parsed = parseURL(url);
-  const importSymbol = camelize(parsed?.package ?? "mod");
-  const importStatement = `import * as ${importSymbol} from "${url}";\n`;
+  const itemParts = item?.split(".");
+  // when the imported symbol is a namespace import, we try to guess at an
+  // intelligent camelized name for the import based on the package name.  If
+  // it is a named import, we simply import the symbol itself.
+  const importSymbol = itemParts
+    ? itemParts[0]
+    : camelize(parsed?.package ?? "mod");
+  // when using a symbol from an imported namespace exported from a module, we
+  // need to create the local symbol, which we identify here.
+  const usageSymbol = itemParts && itemParts.length > 1
+    ? itemParts.pop()
+    : undefined;
+  // if it is namespaces within namespaces, we simply re-join them together
+  // instead of trying to figure out some sort of nested restructuring
+  const localVar = itemParts?.join(".");
+  // we create an import statement which is used to populate the copy paste
+  // snippet of code.
+  let importStatement = item
+    ? `import ${isType ? "type " : ""}{ ${importSymbol} } from "${url}";\n`
+    : `import * as ${importSymbol} from "${url}";\n`;
+  // if we are using a symbol off a imported namespace, we need to destructure
+  // it to a local variable.
```

Comment:
It'd be nice to extract this logic so it can be unit tested and tweaked independently of the component.

Reply:
Yeah, let me refactor it out.